recipe/session/recipeImplementation.go

```diff
@@ -141,7 +139,6 @@ func makeRecipeImplementation(querier supertokens.Querier, config sessmodels.Typ
 			// we clear cookies if it is TokenTheftDetectedError
 			if (defaultErrors.As(err, &errors.UnauthorizedError{}) && (err.(errors.UnauthorizedError).ClearCookies == nil || *err.(errors.UnauthorizedError).ClearCookies)) || defaultErrors.As(err, &errors.TokenTheftDetectedError{}) {
 				supertokens.LogDebugMessage("refreshSession: Clearing cookies because of UNAUTHORISED or TOKEN_THEFT_DETECTED response")
-				clearSessionFromCookie(config, res)
 			}
```

Comment:
You can also clear cookies in the handleError in session recipe in case of token theft detected + logging it, and remove this if statement fully